proxylib/proxylib.go

```diff
@@ -106,20 +106,18 @@ func OnNewConnection(instanceId uint64, proto string, connectionId uint64, ingre
 }
 
 // Skip bytes in input, or exhaust the input.
-// If input is exhausted 'bytes' will return the number of bytes remaining beyond the input
-func advanceInput(bytes, unit, offset int, data *[][]byte) (int, int, int) {
-	for bytes > 0 && unit < len(*data) {
-		rem := len((*data)[unit]) - offset // this much data left in unit
-		if bytes < rem {                   // more than 'bytes' bytes in unit
-			offset += bytes
+func advanceInput(input [][]byte, bytes int) [][]byte {
+	for bytes > 0 && len(input) > 0 {
+		rem := len(input[0]) // this much data left in the first slice
+		if bytes < rem {
+			input[0] = input[0][bytes:] // skip 'bytes' bytes
```

Comment:
You're modifying the input. I suppose that's intentional?

Reply:
The first slice of input is resliced, the bytes underneath are not changed.